### PR TITLE
[ENG-22249] Add servicemonitor

### PR DESCRIPTION
**Pull request by shelbyspees**
Also rename the port for the collector metrics endpoint and remove the extraneous prometheus-exporter endpoint for now. We may add it back when we start using the spanmetrics connector.

### CODE REVIEW

**Review by nshalman — COMMENTED** [review]
Generally makes sense to me. If you've tested it, I'm happy to approve, but I don't fully trust my mental model of K8s to match reality when it comes to the finer details of this change.

File: templates/service.yaml [hunk 1]
```diff
@@ -20,11 +20,25 @@ spec:
...
+  name: opentelemetry-collector-servicemonitor
+  labels:
+    app: opentelemetry-collector
+spec:
+  selector:
+    matchLabels:
+      app: opentelemetry-collector
+  endpoints:
+  - port: prometheus-collector-metrics
+    interval: 15s
```
Comment (lmm) [inline templates/service.yaml:43]: 15s seems like a reasonable interval but WDYT about making this overridable?

**Review by lmm — COMMENTED** [review]
Looks great! added a couple of comments

File: templates/service.yaml (hunk 1 above)
Comment (lmm) [reply templates/service.yaml:43]: We just talked about this, the value is used internally by the collector and matches its own scrape interval.

**Review by nshalman — APPROVED** [review]

**Review by lmm — APPROVED** [review]

**Pull request merged**